src/lib.rs

```diff
@@ -71,6 +71,13 @@ impl Default for Client {
             connector
         });
 
+        #[cfg(feature = "rustls")]
```

Comment:
I think we should explicitly specify the priority of features (both in the code and readme) if both are enabled (because they can be enabled in different parts of the resulting program):
```rust
#[cfg(all(feature = "rustls", not(feature = "tls")))]
```